app/homepage.py

```diff
@@ -14,73 +14,131 @@
+# This function creates the button that allows users to change the map style
+def build_map_style_button():
+
+    button = html.Button(children='Afficher les niveaux de risques',
+                         id='map_style_button')
+
+    return html.Center(button)
+
+
+# This function takes as input the number of clicks on the button defined above and returns the layer style to use
+def choose_map_style(n_clicks):
+
+    # Because we start with the alerts map, if the number of clicks is even, this means that
+    # we are still using the alerts map and we may want to risk one
+    if n_clicks % 2 == 0:
+        button_content_map = 'Afficher les niveaux de risques'
+        map_object = build_alerts_map()
+        slider = ' '
+
+    # If the number of clicks is odd, this means we are using the risks map and may
+    # want to come back to alerts one
+    else:
+        button_content_map = 'Revenir à la carte Alertes'
+        map_object = build_risks_map()
+        slider = build_opacity_slider()
+
+    return button_content_map, map_object, slider
+
+
+# This function either displays or hides meteo graphs from graphs.py
+def meteo_graphs(display=False):
+    if display is True:
+        return dbc.Row(
+            [
+                dbc.Col([html.H2("Données météorologiques"), dcc.Graph(figure=meteo_fig)],
+                        md=4),
+                dbc.Col([html.H2("another indicator"),
+                        dcc.Graph(figure={"data": [{"x": [1, 2, 3], "y": [1, 4, 9]}]})],
+                        md=4),
+                dbc.Col([html.H2("a third indicator"),
+                        dcc.Graph(figure={"data": [{"x": [1, 2, 3], "y": [1, 4, 9]}]})],
+                        md=4),
+            ])
+    else:
+        return ''
+
+
+# ------------------------------------------------------------------------------
+# Content and App layout
 
-# instantiating the map object from alerts.py: to be modified depending on
-# what we expect the user to view first on homepage
+# Instantiating the alert map object from alerts.py and setting it a the default map object:
 map_object = build_alerts_map()
 
-# instantiating fig objects from graphs.py functions
+# Instantiating fig objects from graphs.py functions
 meteo_fig = generate_meteo_fig()
 
+# Instantiating navbar object from navbar.py
 nav = Navbar()
 
-body = dbc.Container(
-    [dbc.Row(
-        [dbc.Col(html.H1('Bienvenue sur Pyronear')),
-         dbc.Col(dcc.Dropdown(id='user_department_input',
-                              options=[{'label': 'Ardèche', 'value': 'Ardèche'},
-                                       {'label': 'Gard', 'value': 'Gard'},
-                                       {'label': 'Landes', 'value': 'Landes'}
-                                       ]))
+# Instantiating map layers button object from alters.py
+map_layers_button = build_layer_style_button()
```

Comment:
This comment is more of a question I am asking myself than a change request. Would it be cleaner to directly call the functions instead of their output when building the `body` object? 

This might not be clear said this way but so far, we generally create a new variable (here `map_layers_button`) which simply contains the output of the related function (`build_layer_style_button`) and then, we call it without any particular transformation in the app layout block (at line 123 in this case). Equivalently, we could directly call the function at line 123 and skip the creation of the object. 

I don't know what the best practice is here and I don't have any strong opinion, but I guess that we would spare quite a few lines and variables by doing so, which can only make our code more readable, right?

Reply:
Totally agreed, to be honest i followed this 'instantiation logic' to ensure code unity with respect to what has been done so far :)

I think that doing a proper PR for this refactoring task would be a good way to deal with it right ?

Reply:
Yes, I guess this logic is inherited from the code initially provided with the tutorials. And fully agreed with the idea of a separate PR!